### ISSUE TITLE
Fix compatibility with psycopg

### ISSUE THREAD

**Issue by FinalAngel** [django-cms/django-cms#6666]
### Summary
Psycopg 2.8 [has just been released](http://initd.org/psycopg/articles/2019/04/04/psycopg-28-released/) and breaks compatibility in django CMS.

### Expected behaviour
When building a project with latest django CMS the build should succeed.

### Actual behaviour
When building a project with latest django CMS the build is failing.

### Other
The issue has been resolved for 3.4, 3.5 and 3.6 on Divio Cloud. Manually pinning **psycopg<2.8** will fix the issue in a project.

**Comment by jrief** [comment]
To be more precise, I stumbled across this error as well.

It happens during this migration:
```
Applying cms.0019_set_pagenode...
line 68, in apply
    connection.introspection.get_table_description(connection.cursor(), 'cms_page')
  File "lib/python3.6/site-packages/django/db/backends/postgresql/introspection.py", line 87, in get_table_description
    )) for line in cursor.description
  File "lib/python3.6/site-packages/django/db/backends/postgresql/introspection.py", line 87, in <listcomp>
    )) for line in cursor.description
TypeError: sequence index must be integer, not 'slice'
```

**Comment by JustDevZero** [comment]
I confirm that this behaviour happens, pinning<2.8 'workarounds', but having 2 years after the last major release, I thing that the migrations should be fixed in order to support the new version.

**Comment by DmytroLitvinov** [comment]
The same behaviour.
Thanks for submitting issue.

**Comment by v582ijay** [comment]
Same here too. Is there any workaround ?

**Comment by JustDevZero** [comment]
As we already said:

Pin psycopg2==2.7.7 in your requirement.txt and run a pip install
psycopg2==2.7.7

**Comment by MARUTHESH** [comment]
Same here.

**Comment by yakky** [comment]
Just FYI this is a Django bug solved in 2.2+ only (see https://code.djangoproject.com/ticket/30331)

The only solution (documented by django itself) for 2.1 and below is using psycopg2 2.7.x as reported here. There is no available fix in django CMS

**Comment by Aiky30** [comment]
Closing as < Django 2.2 is no longer supported. Resolved in Django 2.2+
